docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -128,3 +134,177 @@ class CustomerReader extends SprykerCustomerReader implements CustomerReaderInte
+{
+   /**
+     * 
+     */
+    public function setContentTypesToAccessible(CustomerAccessTransfer $customerAccessTransfer): void
+    {
+        $this->getFactory()->createCustomerAccessMapper();
+       ....
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Please avoid usage of Spryker\...\CustomerAccessEntityManager::createCustomerAccessMapper(...) in Pyz/Zed/CustomerAccess/Persistence/CustomerAccessEntityManager
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+See `Resolving the error` section for the first example on this page.
+
+### Example of resolving the error by cloning functionality to the project level
+
+Copy the method to the factory on the project level.
+
+```php
+namespace Pyz\Zed\Customer\Business;
+
+use Pyz\Zed\Customer\Business\Customer\CustomerReader;
+use Spryker\Zed\Customer\Business\CustomerBusinessFactory as SprykerCustomerBusinessFactory;
+
+class CustomerBusinessFactory extends SprykerCustomerBusinessFactory
+{
+    /**
+     * @return \Pyz\Zed\Customer\Business\Customer\CustomerReaderInterface
+     */
+    public function createPyzCustomerAccessMapper(): CustomerReaderInterface
+    {
+        return new CustomerPyzReader(...);
+    }
+}
+```
+
+
+## PrivateApi:Dependency
+
+Business factory should use dependency by a key that is defined on project level.
+
+### Example of code that causes an upgradability error
+
+`CustomerAccessBusinessFactory` uses as key `ProductPageSearchDependencyProvider::QUERY_CONTAINER_CATEGORY` that was declared on the vendor level.
+
+```php
+...
+use Spryker\Zed\ProductPageSearch\ProductPageSearchDependencyProvider
+...
+class CustomerAccessBusinessFactory implements SprykerCustomerAccessBusinessFactory
+{
+    /**
+    * @return \Pyz\Zed\CustomerAccess\Business\PropelFacadeInterface
+    * @throws \Spryker\Zed\Kernel\Exception\Container\ContainerKeyNotFoundException
+    */
+    public function getPropelFacade(): PropelFacadeInterface
+    {
+        return $this->getProvidedDependency(ProductPageSearchDependencyProvider::QUERY_CONTAINER_CATEGORY);
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Please avoid usage of Spryker/Zed/ProductPageSearch/ProductPageSearchDependencyProvider/ProductPageSearchDependencyProvider::QUERY_CONTAINER_CATEGORY in Pyz/Zed/CustomerAccess/Business/CustomerAccessBusinessFactory
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+See `Resolving the error` section for the first example on this page.
```

Comment:
Provide the link on all places where you redirect the user of the documentation.

Text should be also changed:

```
Solution is the error is the same as in `Resolving the error`[#link-on-something]
```

Reply:
Changed to Solution is the error is the same as in [Resolving the error](#privateapi:facade)